### DISCUSSION TITLE
When the fields are changed for the first time, all fields are re-rendered

### DISCUSSION THREAD

**Discussion by ya-tmch** [react-hook-form/react-hook-form#7351] [General]
Why in this example, when the fields are changed for the first time, all fields are re-rendered?

https://codesandbox.io/s/react-hook-form-example-forked-zkrb0?file=/src/index.js

**Comment by bluebill1049** [comment]
because you are subscribed to isDrity form state at the root of your application. you can consider to use that https://react-hook-form.com/api/useformstate

**Reply by ya-tmch** [reply]
Thanks for the answer!

It turns out that the flag is emits rerender even if it does not change its state (always true after the first change is entered)?

**Reply by ya-tmch** [reply]
And why are the inputs re-rendered if there is no subscription to the form state? Thanks
